app/Helper/ConfigurationParser.php

```diff
@@ -4,35 +4,160 @@
 namespace ArtifactCreation\Helper;
 
 
-use Exception;
+use ArtifactCreation\Builder\PackageConfigurationBuilder;
+use ArtifactCreation\Exception\InvalidFileException;
+use ArtifactCreation\Exception\MissingConfigurationException;
+use ArtifactCreation\Exception\MissingFileException;
+use ArtifactCreation\Exception\MissingParameterException;
+use ArtifactCreation\Model\ComposerConfiguration;
+use ArtifactCreation\Model\PackageConfigurationAbstract;
 
+/**
+ * Parses the mandatory configuration file
+ * and acts as a DTO
+ *
+ * Class ConfigurationParser
+ * @package ArtifactCreation\Helper
+ */
 class ConfigurationParser
 {
+    /** @var string CONFIGURATION_KEY_COMPOSER */
     const CONFIGURATION_KEY_COMPOSER = 'composer';
+    /** @var string CONFIGURATION_KEY_PACKAGE */
     const CONFIGURATION_KEY_PACKAGE = 'package';
+    /** @var string CONFIGURATION_KEY_PACKAGE_TYPE */
+    const CONFIGURATION_KEY_PACKAGE_TYPE = 'type';
+
+    /** @var array $configuration */
+    protected $configuration;
+    /** @var ComposerConfiguration|null $composerConfiguration */
+    protected $composerConfiguration;
+    /** @var PackageConfigurationAbstract $packageConfiguration */
+    protected $packageConfiguration;
+
 
     /**
+     * ConfigurationParser constructor.
+     *
      * @param $fullyQualifiedFileName
+     *
+     * @throws InvalidFileException
+     * @throws MissingConfigurationException
+     * @throws MissingFileException
+     * @throws MissingParameterException
+     */
+    public function __construct($fullyQualifiedFileName)
+    {
+        $this->configuration = $this->parseJsonFile($fullyQualifiedFileName);
+        $this->setPackageConfiguration();
+        $this->setComposerConfiguration();
+    }
+
+
+    /**
+     * Parse the json file with the given fully qualified filename
+     *
+     * @param $fullyQualifiedFileName
+     *
      * @return mixed
-     * @throws Exception
+     *
+     * @throws InvalidFileException
+     * @throws MissingFileException
      */
-    public static function parseJsonFile($fullyQualifiedFileName)
+    protected function parseJsonFile($fullyQualifiedFileName)
     {
         $fileContentRaw = file_get_contents($fullyQualifiedFileName);
         if ($fileContentRaw === false) {
-            throw new Exception('Could not load configuration file!');
+            throw new MissingFileException('Could not load configuration file!');
+        }
+
+        $fileContent = json_decode($fileContentRaw, true);
```

Comment:
1. May be call result json_decode more consistant to what we get?
2. By contract it should be always array as return value. -> Check for is_array